controller_interface/include/controller_interface/controller_interface.hpp

```diff
@@ -134,6 +135,56 @@ class ControllerInterface : public rclcpp_lifecycle::node_interfaces::LifecycleN
     }
   }
 
+  // Methods for chainable controller types with default values so we can put all controllers into
+  // one list in Controller Manager
+
+  /// Get information if a controller is chainable.
+  /**
+   * Get information if a controller is chainable.
+   *
+   * \returns true is controller is chainable and false if it is not. \default false if controller
+   * implements ControllerInterface.
+   */
+  CONTROLLER_INTERFACE_PUBLIC
+  virtual bool is_chainable() const { return false; }
```

Comment:
is it an absolute *must* to have the chainable API spill over to the top level controller interface?

Reply:
Yes please.

Also let's keep the discussion up re. spilling the API over. I think this is a distinct enough feature to justify having a second vector for chainable controllers, no?

Reply:
I would have to try to implement this to actually know. It would cause some code duplication and complexities into CM, but maybe would not be so bad as I think